components/board/genericlinux/data.go

```diff
@@ -96,16 +97,22 @@ func GetGPIOBoardMappings(modelName string, boardInfoMappings map[string]BoardIn
 // getCompatiblePinDefs returns a list of pin definitions, from the first BoardInformation struct
 // that appears compatible with the machine we're running on.
 func getCompatiblePinDefs(modelName string, boardInfoMappings map[string]BoardInformation) ([]PinDefinition, error) {
-	const compatiblePath = "/proc/device-tree/compatible"
+	var path string
+	const (
+		compatiblePath    = "/proc/device-tree/compatible"
+		compatiblePathDmi = "/sys/devices/virtual/dmi/id/board_name"
+	)
+
+	if modelName == "intel" {
+		path = compatiblePathDmi
+	} else {
+		path = compatiblePath
+	}
 
-	compatiblesRd, err := os.ReadFile(compatiblePath)
+	compatibles, err := newStringSetFromFile(modelName, path)
```

Comment:
I guess this works, but the helper function I would have created is `getCompatibles`. Something like this:

```
func getCompatibles(modelName string) []string, err {
    const (
        compatiblePathDt = "/proc/device-tree/compatible"
        compatiblePathDmi = "/sys/devices/virtual/dmi/id/board_name"
    )
    if compatiblePathDmi is a valid file:
        return a length-1 StringSet containing its contents

    read compatiblePathDt, return its contents split on null bytes
}
```

Making the `path` variable have two different purposes and passing that around seems like extra mental overhead for anyone working on this code, and then trying to write a single implementation of `newStringSetFromFile` that supports both device-tree and DMI boards seems hard.